sympy/physics/mechanics/tests/test_joint.py

```diff
@@ -373,7 +373,7 @@ def test_slidingjoint_arbitrary_axis():
     N, A, P, C = _generate_body()
     PrismaticJoint('S', P, C, child_axis=-A.x)
 
-    assert -A.x.angle_between(N.x) == -pi
+    assert (-A.x).angle_between(N.x) == 0
```

Comment:
This is a fix for this test, but isn't a fix for the method. `angle_between()` doesn't behave as expected. In general with sympy objects you'd expect `-x.something()` to operate on `-x` not `x`. Is this an issue because `.x` is an attribute of `A`?

An issue should be opened about `angle_between()` not respecting the leading negative sign. And, for now, `angle_between()` needs an example like this in its doc string, otherwise users will be caught by this unexpected issue.

Reply:
I think this problem is not limited to `angle_between()`, all member functions of class `Vector` are ignoring the leading negative signs.

Reply:
Maybe this only incidentally works:

```
In [1]: import sympy.physics.mechanics as me

In [3]: A = me.ReferenceFrame('A')

In [4]: -A.x.cross(A.y)
Out[4]: - A.z

In [5]: A.x.cross(A.y)
Out[5]: A.z
```

Reply:
If there is no way to make this work correctly without the parentheses, then we need to make a clear "gotcha" statement in the documentation of `angle_between()`, because it will give the incorrect values. We also need to make sure the use of `angle_between()` in `Joint` can't be caught by this issue in an edge case.

Reply:
Yes I made sure, `angle_between` in `Joint` would always be correct.